server/pulp/server/webservices/views/search.py

```diff
@@ -0,0 +1,118 @@
+"""
+This module contains search View superclasses. Your code should subclass these to achieve maximum
+searching capability.
+"""
+from django.views import generic
+
+from pulp.server import exceptions
+from pulp.server.auth import authorization
+from pulp.server.db.model import criteria
+from pulp.server.webservices.controllers.decorators import auth_required
+from pulp.server.webservices.views import util
+
+
+class SearchView(generic.View):
+    """
+    This class is meant to be subclassed by views that need to provide search functionality on Pulp
+    models. In most cases, it should be possible for the subclasses to define only a few class
+    variables. Subclasses must define at least one of these two class attributes:
+
+    0) If the SearchView is for searching a MongoEngine model, model should be defined.
+    1) If it's for an "old-style" model, manager must be defined.
+
+    :cvar    response_builder: The function that should be used to turn the search results
+                               into a JSON serialized Django Response object. If not defined,
+                               this defaults to
+                               pulp.server.webservices.views.util.generate_json_response.
+    :vartype response_builder: staticmethod
+    :cvar    manager:          Define this class attribute if you are making a SearchView for
+                               a model that has not yet been converted to MongoEngine. It
+                               should represent an instance of the Model's manager class,
+                               which must have a find_by_criteria() method.
+    :vartype manager:          object
+    :cvar    model:            Define this class attribute if you are making a SearchView for
+                               a MongoEngine Document. The model must have a meta class
+                               attribute defined with the 'queryset_class' key indexing
+                               pulp.server.db.model.base.CriteriaQuerySet, so that the
+                               Document's QuerySet will have a find_by_criteria() method.
+    :vartype model:            mongoengine.Document
+    :cvar    serializer:       If your view needs to modify the QuerySet results before they
+                               are returned to the caller, you can define this attribute. It
+                               should be set to a function that accepts a single object, and
+                               returns a single object.
+    :vartype serializer:       staticmethod
+    """
+    response_builder = staticmethod(util.generate_json_response)
```

Comment:
In our testing, this is why response_builder worker, but serializer couldn't be found. You were seeing this response_builder.